docs/dev-guide.md

```diff
@@ -0,0 +1,22 @@
+# Development Guide
+
+## Overview
```

Comment:
This all looks useful to have, but I think probably the first part should focus on the tools you need setup to develop, the scripts to generate code, etc., and what that workflow looks like, before we dive into details of the code.